### DISCUSSION TITLE
Could 'Search Likes' And 'Limit Posts Shown From A Single User' Be Added?

### DISCUSSION THREAD

**Discussion by RedBreloom** [AprilSylph/XKit-Rewritten#405] [Suggestion Box]
My likes are super large and go back years and it'd be really useful to be able to search them by keyword or tag again. Though I know with the old version it got pretty laggy if too much was searched though.
And the other one I don't remember what its precise name was but it was 'the one to hide x amount of posts from one user if they've posted a ton at once.' Like when one user posts like 20 posts in a row and you just want to see other users' posts already, this would hide them all after say, 4.
Are these feasible to be added to Xkit Rewritten?

**Comment by AprilSylph** [comment]
Limit People is on [the to-do list](https://github.com/AprilSylph/XKit-Rewritten/projects/4). This project is still in progress, and former features of New XKit are being added in order of least complexity first.

I don't plan to bring Search Likes back, at least not yet. The way Search Likes worked in New XKit was basically just to hide liked posts not containing the search term, forcing Tumblr to load more and more likes until there were enough results on-screen (or maybe until there were no more likes to load, not sure). So, you can essentially achieve the same effect by using Scroll to Bottom to load all your likes, and then a keyword search in your browser (CTRL+F or CMD+F) for what you're looking for.

**Reply by RedBreloom** [reply]
Oh awesome, missed that one. I will look forward to it eventually being included.

And to follow up on the search likes method. It makes sense that search likes really lagged my computer before then. I tried the scrolling to the bottom method last night since my current rig is much more capable. it started lagging really badly around several thousand likes. I think I stopped around 10k likes because firefox was giving me warnings about the page slowing down too much. Firefox was using like 7gb of RAM. I have way more likes than 10k like (my account is like 8 years old). So it's definitely totally infeasible on a larger scale with the old method. Hopefully something better comes along some day though.

**Reply by marcustyphoon** [reply]
Yeah, while there may be ways to optimize the resource use of this just-load-everything method somewhat (I've done some fun things in testing), fundamentally, I don't know if we really want to facilitate requesting 10000 posts from Tumblr's servers super easily :D I don't actually know if someone has submitted "the ability to search our likes" to https://www.tumblr.com/support; I would suggest doing so politely in the off chance Staff thinks it's worth implementing on the server side! Of course, Tumblr's built-in search function is uh... a bit... it's not the most reliable thing, yet, in my experience. But better than nothing.

I think the way I would want it to work ideally is something more like "download your Tumblr data as an archive and search through that archive locally." That way, each of your liked posts only has to be downloaded from Tumblr once no matter how much you want to search for at any time in the future.

**Reply by RedBreloom** [reply]
That sounds like it'd be a good alternative way to do it tbh. Making a searchable archive I mean. Suggesting it to staff isn't a bad idea either.
Also unrelated question but has tumblr's general search changed more recently? I notice 'top of all time' searches return mostly stuff from this year or last year now. And that the searches are pretty much identical between 'search' and 'tagged' variants of the url. I'm just finding it annoying to find older or more diverse content on tumblr lately so I wanted to ask. I'm not sure if there's anything to do for it, or if it's something that I'd need to contact tumblr staff about.

**Reply by marcustyphoon** [reply]
Re: search changes, I think I have seen the topic discussed, but I personally don't know!

**Reply by RedBreloom** [reply]
All right, I just sent tumblr support a message on that then.
One more thing, could the 'related tags' and 'related blogs' sections of search pages be added to the 'no reccomends' app or something else? If those could be removed it'd be much easier for me to view search results in 3 columns instead of two like it was previously.

**Reply by marcustyphoon** [reply]
You're not the only one to request this! I'm not sure if it's actually possible from XKit's side, though, as I believe Tumblr's code checks the browser page width to decide how many columns to put the posts in. I'll write it up as its own thread, but we may need to ask Staff if they would be interested in implementing a toggle for this.

**Reply by AprilSylph** [reply]
Yep, simple CSS hacks would not increase the number of columns. Hiding both the related tags and related blogs wouldn't hide the sidebar automatically, but even when it is hidden, the main container doesn't grow to fill the space. And, even forcing the main container to grow, Tumblr controls the masonry, which ends up just putting more blank space either side of the posts columns.
